modules/Bio/EnsEMBL/Compara/RunnableDB/HomologyAnnotation/DiamondBlastp.pm

```diff
@@ -0,0 +1,94 @@
+
+     http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+
+=head1 NAME
+
+Bio::EnsEMBL::Compara::RunnableDB::HomologyAnnotation::DiamondBlastp
+
+=head1 DESCRIPTION
+
+Create fasta file containing batch_size number of sequences. Run DIAMOND blastp and parse
+the output into PeptideAlignFeature objects. Store PeptideAlignFeature objects in the compara
+database
+
+=cut
+
+package Bio::EnsEMBL::Compara::RunnableDB::HomologyAnnotation::DiamondBlastp;
+
+use strict;
+use warnings;
+
+use base ('Bio::EnsEMBL::Compara::RunnableDB::BlastAndParsePAF');
+
+sub get_queries {
+    my $self = shift @_;
+
+    my $start_member_id = $self->param_required('start_member_id');
+    my $end_member_id   = $self->param_required('end_member_id');
+
+    #Get list of members and sequences
+    my $member_ids = $self->compara_dba->get_HMMAnnotAdaptor->fetch_all_seqs_missing_annot_by_range($start_member_id, $end_member_id, 'no_null');
+    return $self->compara_dba->get_SeqMemberAdaptor->fetch_all_by_dbID_list($member_ids);
+}
+
+sub run {
+    my $self = shift @_;
+
+    my $diamond_exe           = $self->param('diamond_exe');
+    my $blast_params          = $self->param('blast_params')  || '';  # no parameters to C++ binary means having composition stats on and -seg masking off
+    my $evalue_limit          = $self->param('evalue_limit');
+    my $worker_temp_directory = $self->worker_temp_directory;
+    my $blast_outfile         = $worker_temp_directory . '/blast.out.' . $$; # looks like inevitable evil
+
+    Bio::EnsEMBL::Compara::Utils::Preloader::load_all_sequences($self->compara_dba->get_SequenceAdaptor, undef, $self->param('query_set'));
+
+    if ($self->debug) {
+        my $blast_infile  = $worker_temp_directory . '/blast.in.' . $$;  # only for debugging
+        print "diamond_infile $blast_infile\n";
+        my $members = $self->param('query_set')->get_all_Members;
+        $self->param('query_set')->print_sequences_to_file($blast_infile, -format => 'fasta');
+    }
+
+    $self->compara_dba->dbc->disconnect_if_idle();
+
+    my $cross_pafs = [];
+    foreach my $blast_db (keys %{$self->param('all_blast_db')}) {
+        my $target_genome_db_id = $self->param('all_blast_db')->{$blast_db};
+
+        my $cmd = "$diamond_exe blastp -d $blast_db --evalue $evalue_limit --out $blast_outfile --outfmt 6 qseqid sseqid evalue score nident pident qstart qend sstart send length positive ppos qseq_gapped sseq_gapped $blast_params";
+
+        my $run_cmd = $self->write_to_command($cmd, sub {
+            my $blast_fh = shift;
+            $self->param('query_set')->print_sequences_to_file($blast_fh, -format => 'fasta');
+        } );
```

Comment:
Can the input file be given to diamond on the command-line ? If so, it is more reliable to create the input file (can remove the `if ($self->debug)` line 59), pass it to diamond and use `$self->run_command`

Reply:
Yes it can, I did it this way because this was the way it was with `BLASTp` originally and assumed there was a good reason to do so, I much prefer passing the input file.